pulpcore/cli/deb/context.py

```diff
@@ -10,12 +13,74 @@
 _ = gettext.gettext
 
 
+class PulpAptDistributionContext(PulpEntityContext):
+    ENTITY = _("apt distribution")
+    ENTITIES = _("apt distributions")
+    HREF = "deb_apt_distribution_href"
+    LIST_ID = "distributions_deb_apt_list"
+    READ_ID = "distributions_deb_apt_read"
+    CREATE_ID = "distributions_deb_apt_create"
+    UPDATE_ID = "distributions_deb_apt_partial_update"
+    DELETE_ID = "distributions_deb_apt_delete"
+
+    def preprocess_body(self, body: EntityDefinition) -> EntityDefinition:
+        body = super().preprocess_body(body)
+        # Unsure if deb needs this check
+        if self.pulp_ctx.has_plugin(PluginRequirement("core", min="3.16.0.dev")):
+            if "repository" in body and "publication" not in body:
+                body["publication"] = None
+            if "repository" not in body and "publication" in body:
+                body["repository"] = None
+        return body
+
+
+class PulpAptPublicationContext(PulpEntityContext):
+    ENTITY = _("apt publication")
+    ENTITIES = _("apt publications")
+    HREF = "deb_apt_publication_href"
+    LIST_ID = "publications_deb_apt_list"
+    READ_ID = "publications_deb_apt_read"
+    CREATE_ID = "publications_deb_apt_create"
+    DELETE_ID = "publications_deb_apt_delete"
+
+    def preprocess_body(self, body: EntityDefinition) -> EntityDefinition:
+        body = super().preprocess_body(body)
+        version = body.pop("version", None)
+        if version is not None:
+            repository_href = body.pop("repository")
+            body["repository_version"] = f"{repository_href}versions/{version}/"
+        return body
+
+
+class PulpVerbatimPublicationContext(PulpEntityContext):
+    ENTITY = _("verbatim publication")
+    ENTITIES = _("verbatim publications")
+    HREF = "deb_verbatim_publication_href"
+    LIST_ID = "publications_deb_verbatim_list"
+    READ_ID = "publications_deb_verbatim_read"
+    CREATE_ID = "publications_deb_verbatim_create"
+    DELETE_ID = "publications_deb_verbatim_delete"
+    APT_ONLY: ClassVar[Iterable[str]] = ("simple", "structured", "signing_service")
+
+    def preprocess_body(self, body: EntityDefinition) -> EntityDefinition:
+        body = super().preprocess_body(body)
+        for field in self.APT_ONLY:
+            body.pop(field, None)
```

Comment:
I'm still not convinced by this. But at least we should not silently drop these parameters.

Reply:
Should this wait for polymorphic options? I can get it ready for review so it can be in the next cli release.

Reply:
No, i'm just saying we should error on unsupported options, because the user intended them to do something the cli is not gonna carry out.